File: 0611-valid-triangle-number/0611-valid-triangle-number.java
class Solution {
    public int triangleNumber(int[] nums) {
       Arrays.sort(nums);
       int count = 0;
       int n = nums.length;
       //Check all triplets using 3 nested loop
       for(int i=0;i<n;i++){
         for(int j = i +1;j<n-1;j++){
            for(int k = j+1;k<n;k++){
                //After sorting we need to check a+b>c
                if(nums[i] + nums[j]>nums[k]){
                    count++;
                }else{
                    break;
                }
            }
         }
       }
       return count;
    }
}

File: 0611-valid-triangle-number/0611-valid-triangle-number.py
class Solution(object):
    def triangleNumber(self, nums):
        nums.sort()
        n = len(nums)
        count = 0

        for k in range(n - 1, 1, -1):  # fix largest side
            i = 0
            j = k - 1

            while i < j:
                if nums[i] + nums[j] > nums[k]:
                    count += (j - i)  # count all valid pairs
                    j -= 1
                else:
                    i += 1

        return count